pkg/netceptor/netceptor.go

```diff
@@ -441,6 +441,27 @@ func (s *Netceptor) MaxConnectionIdleTime() time.Duration {
 	return s.maxConnectionIdleTime
 }
 
+// Sets the MaxConnectionIdleTime object on the Netceptor instance
+func (s *Netceptor) SetMaxConnectionIdleTime(userDefinedMaxIdleConnectionTimeout string) error {
+	// before we instantiate a new instance of Netceptor, let's verify that the user defined maxidleconnectiontimeout value is parseable
+	duration, err := time.ParseDuration(userDefinedMaxIdleConnectionTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to parse MaxIdleConnectionTimeout from configuration file -- valid examples include '1.5h', '30m', '30m10s'")
+	}
+	// we don't want the user defined timeout to be less than the defaultMaxConnectionIdleTime constant
+	if duration < defaultMaxConnectionIdleTime {
+		return fmt.Errorf("user defined maxIdleConnectionTimeout [%d] is less than the default default timeout [%d]", duration, defaultMaxConnectionIdleTime)
+	}
+	// we also need to ensure that if defined, a user defined connection timeout is at least 2x longer than the defaultRouteUpdateTime value
+	if duration < (2*defaultRouteUpdateTime + 1*time.Second) {
```

Comment:
I think this check is unnecessary, given that `defaultMaxConnectionIdleTime` is defined as 2x defaultRouteUpdateTime + 1 second, and we ensure that duration < defaultMaxConnectionIdleTime in the previous check.